test/test.mjs

```diff
@@ -1,59 +1,59 @@
+
 const args = {
   seconds: 0,
-  executablePath: process.argv[2] || '/usr/bin/brave',
+  executablePath: browserPath,
   adblockLists: {
     eaokkjgnlhceblfhbhpeoebmfldocmnc: false,
     adcocjohghhfpidemphmcmlmhnfgikei: false,
     cdbbhgbmjhfnhnmgeddbliobbofkgdhe: false,
     bfpgedeaaibpoidldhjcknekahbikncb: false
   }
 }
+// Calculate concurrency based on available CPU cores
+const CONCURRENCY = Math.max(1, Math.floor(cpus().length / 2))
+console.log(`Running tests with concurrency: ${CONCURRENCY}`)
 
-async function testPage (testCasePath, expectedHash, expectedScrollBlocking) {
+async function testPage (t, testCasePath, expectedHash, expectedScrollBlocking) {
   const url = pathToFileURL(path.join(import.meta.dirname, 'data', testCasePath, 'index.html')).href
-  return checkPage({ url, hostOverride: testCasePath, blockNonHttpRequests: false, ...args }).then(r => {
-    if (r.error) {
-      console.log('[' + testCasePath + '] ERROR: ' + r.error)
-      return false
-    }
+  const r = await checkPage({ url, hostOverride: testCasePath, blockNonHttpRequests: false, ...args })
 
-    const failures = []
-    const warnings = []
+  if (r.error) {
+    throw new Error(`[${testCasePath}] ERROR: ${r.error}`)
+  }
 
-    let markupHash
-    if (r.identified) {
-      markupHash = createHash('sha256').update(r.markup).digest('base64')
-    }
+  let markupHash
+  if (r.identified) {
+    markupHash = createHash('sha256').update(r.markup).digest('base64')
+  }
 
-    if (expectedHash !== markupHash) {
-      failures.push('expected cookie notice hash "' + expectedHash + '" did not match detected markup "' + markupHash + '"')
-    }
+  const cookieNoticeTestName = expectedHash === undefined
+    ? 'should not detect notice'
+    : 'should detect notice'
 
-    if (expectedScrollBlocking === undefined) {
-      // once more scroll blocking testcases have been gathered, it should
-      // be possible to improve the heuristics and remove the warnings
-      warnings.push('scroll blocking test ignored')
-    } else if (expectedScrollBlocking !== r.scrollBlocked) {
-      failures.push(`expected scroll blocking result [${expectedScrollBlocking}] did not match detected result [${r.scrollBlocked}]`)
-    }
+  await t.test(cookieNoticeTestName, async (t) => {
+    t.assert.strictEqual(markupHash, expectedHash,
+      `expected hash "${expectedHash}" did not match markup "${markupHash}"`)
+  })
 
-    const success = failures.length === 0
-    const testStatus = success ? 'success' : 'failure'
-    console.log(`[${testCasePath}] ${testStatus}`)
-    for (const failure of failures) {
-      console.log(` - ${failure}`)
-    }
-    for (const warning of warnings) {
-      console.log(` - ${warning}`)
+  await t.test('scroll blocking detection', async (t) => {
+    if (expectedScrollBlocking === undefined) {
+      // t.diagnostic('scroll blocking test ignored')
+      t.todo('scroll blocking test ignored')
```

Comment:
any reason to choose one over the other? My only personal preference is to choose one and leave the comment out :joy: